Classes/common/CDTFetchChanges.m

```diff
@@ -39,6 +54,13 @@ - (instancetype)initWithDatastore:(CDTDatastore *)datastore
 
 - (void)main
 {
+    // Ensure our settings are not changed during execution
+    self.mDocumentChangedBlock = self.documentChangedBlock;
+    self.mDocumentWithIDWasDeletedBlock = self.documentWithIDWasDeletedBlock;
+    self.mFetchRecordChangesCompletionBlock = self.fetchRecordChangesCompletionBlock;
+    self.mDatastore = self.datastore;
+    self.mStartSequenceValue = self.startSequenceValue;
```

Comment:
On the other hand, I now realize one could write a "completed" block that simply reacts differently based upon the accumulated knowledge received during the 'changes' and 'deleted' blocks... which means there wouldn't be a need to set a different "completion" block. The users's completion block would contain the logic.